packages/core/src/app/customer/getSupportedMethods.ts

```diff
@@ -28,9 +31,10 @@ export const SUPPORTED_METHODS: string[] = [
     'googlepaytdonlinemart',
 ];
 
-export const getSupportedMethodIds = (methodIds: string[]): string[] => {
+export const getSupportedMethodIds = (methodIds: string[], features: CheckoutSettings['features'] = {}): string[] => {
     return methodIds.filter((methodId) => {
-        if (methodId === APPLE_PAY && !isApplePayWindow(window)) {
+        // TODO: this check have to be deleted after implementation of Apple Pay for third party browsers will be tested and released
+        if (methodId === APPLE_PAY && !isApplePayWindow(window) && !features['PAYPAL-4324.applepay_web_browser_support']) {
```

Comment:
My suggestion would be to move the logic of `'PAYPAL-4324.applepay_web_browser_support'` in `isApplePayWindow` method so eventually when this experiment is rolled out we can remove the whole method?

Reply:
This method will be removed in any case when the exp is rolled out, because we use it only for filtering apple pay method. I do not see a valid reason to move this check to `isApplePayWindow` in this case

Reply:
It will be a cleaner and more encapsulated approach where all logic lives in one place.

Reply:
Are you suggesting to pass the features to `isApplePayWindow` and handle this logic there?

Reply:
Why am i asking it is because i do not understand what is difference between checking this exp in current place and checking inside isApplePayWindow if eventually this function and this `getSupportedMethods` method will be deleted, except we're gonna do another nesting?